pandas/__init__.py

```diff
@@ -31,150 +31,153 @@
         "'python setup.py build_ext --force' to build the C extensions first."
     ) from e
 
+# Use redundant imports (X as X) for type checkers to know what is part of the
+# public API. Pandas is not (yet) a py.typed library: the public API is determined
+# based on the documentation.
 from pandas._config import (
-    get_option,
-    set_option,
-    reset_option,
-    describe_option,
-    option_context,
-    options,
+    get_option as get_option,
```

Comment:
wouldn't it be better to simply have a `__all__` ? (i think we actually test the visibility of the names already)

Reply:
Yes, that is possible!

https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface
> A module can expose an __all__ symbol at the module level that provides a list of names that are considered part of the interface. This overrides all other rules above, allowing imported symbols or symbols whose names begin with an underscore to be included in the interface.

Reply:
Replaced the `X as X` with a long `__all__`.

Reply:
Should `pandas.core.config_init` be added to `__all__`? I didn't add it
https://github.com/pandas-dev/pandas/blob/9e9c0de285777453eab68069c1292d0ba4e23f84/pandas/__init__.py#L43

Reply:
umm if you don't need it then no

Reply:
The existing tests in pandas/tests/api/test_api.py do not expect `config_init` to be exported.